### ISSUE TITLE
The implementation of GPT2 masked attention mechanism will cause errors when the model was trained after some iterations.

### ISSUE THREAD

**Issue by xunzi2020** [huggingface/transformers#3181]
# 🐛 Bug

## Information
Model I am using (GPT2):
Language I am using the model on ( Chinese ...): 

File in modeling_gpt2.py：
146        w = w * b - 1e4 * (1 - b)  # here the bias "-1e4" is too big, suggest replace it to "-1e10"

The item value of "query * key" may be close to or lower to "-1e4" ,  then after the softmax operation, the weights of masked attention may be attend to "unseen" context. This will cause inference errors when to evaluate the model.

**Comment by patrickvonplaten** [comment]
Thanks for the bug report @xunzi2020! Did you by any chance measure the degree results would differ when replacing `-1e4` by `-1e10`?

**Comment by xunzi2020** [comment]
Yes, I trained with chinese text (~30GB), it happened.


Patrick von Platen <notifications@github.com> 于2020年3月9日周一 下午4:11写道：

**Comment by patrickvonplaten** [comment]
Do you have a comparison such as: 

| - | GPT2 with -1e4 vs.  GPT2 with -1e10 | 
| ------------- | ------------- | 
| abs mean(softmax(logits) - softmax(logits)) | ???? |
| relative mean(softmax(logits) - softmax(logits)/softmax(logits))  | ???? |

lets say averaged over 100 - 1000 input samples and all GPT2 logits (50256)? 

That would be great to quantify the impact this change would have.
